MementoMori/MementoMori.Server/CardFile/FileWriter.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.IO;
+namespace MementoMori.Server
+{
+    public class FileWriter
+    {
+        private readonly string _directoryPath;
+
+        // Constructor to initialize the directory where files will be stored
+        public FileWriter()
+        {
+            // Get the current directory of the MementoMori.Server project
+            string serverDirectory = Directory.GetCurrentDirectory();
+
+            // Set the path to "CardFile" folde
+            _directoryPath = Path.Combine(serverDirectory, "CardFile");
+
+        }
+
+        // Method to create a file using tags and text
+        public void CreateFile(string tags, string text, string deckId)
+        {
+            // Use the specified file name "001.txt"
+            string fileName = "001.txt";
+            string filePath = Path.Combine(_directoryPath, fileName);
+            Guid cardId = Guid.NewGuid();
+
+            // Validate based on the whole direcotry or smth
+            if(fileName != deckId + ".txt")
+                throw new InvalidOperationException("Error: The deck file does not exist.");
+
+            var fileInfo = new FileInfo(filePath);
+
+            if(fileInfo.Length == 0)
+            {
+                using(StreamWriter sw = File.AppendText(filePath))
+                {
+                    sw.WriteLine("DeckId: " + deckId);
+                    sw.WriteLine("Number of cards: " + 1);
+                    sw.WriteLine("CardIds: " + cardId + "\n");
+                }
+            }
+            else
+            {
+                // Add change to the nr of cards, add extra card id to the line seperated by ';'
+                string[] fileLines = File.ReadAllLines(filePath);
+                fileLines[2] += $"; {cardId}";
+                File.WriteAllLines(filePath, fileLines);
+
+                string[] cardIds = fileLines[2].Substring(fileLines[2].IndexOf(':') + 1).Split(';', StringSplitOptions.RemoveEmptyEntries);
+                int i = 0;
+                foreach(string y in cardIds)
+                {
+                    i++;
+                }
+                fileLines[1] = "Number of cards: " + i;
+                File.WriteAllLines(filePath, fileLines);
+            }
+
+            // Create the content to write to the file
```

Comment:
Should probably serialize and save as json, that way u want need extra `(start)`, `(end)`, etc..

[MS docs](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to  )

Reply:
Wont fix